src/main/java/com/couchbase/client/core/endpoint/AbstractEndpoint.java

```diff
@@ -92,6 +92,14 @@
  */
 public abstract class AbstractEndpoint extends AbstractStateMachine<LifecycleState> implements Endpoint {
 
+    /**
+     * A flag which controls whether the endpoint should force a DNS lookup request for the provided
+     * hostname in case the socket get disconnect.
+     */
+    protected static final boolean FORCE_DNS_LOOKUP_ON_DISCONNECT = Boolean.parseBoolean(
+            System.getProperty("com.couchbase.forceDnsLookupOnDisconnect", "false")
```

Comment:
maybe nitpicking here, but should it be "on reconnect" instead "on disconnect"?

Reply:
Right, I think that would be a better name, considering we do send the DNS request when _reconnecting_.